### PR TITLE
refine fc_infershape

### PR DESCRIPTION
**Pull request by luotao1**
- follow #16154 to refine fc infershape
- add documents for attribute `kAllKernelsMustComputeRuntimeShape `
### commands
`./paddle/fluid/inference/tests/api/test_analyzer_pyramid_dnn --infer_model=third_party/inference_demo/pyramid_dnn/model/ --infer_data=third_party/inference_demo/pyramid_dnn/data.txt --gtest_filter=Analyzer_Pyramid_DNN.profile --repeat=1000 --zero_copy=1 --profile`
### results:
- fc from 35.9336ms to 31.7109ms.
- before:
```
Event                                Calls       Total       Min.        Max.        Ave.        Ratio.
thread0::hash                        12000       71.729      0.004994    0.216792    0.00597742  0.304234
thread0::fused_embedding_seq_pool    16000       63.7859     0.0031      0.11274     0.00398662  0.270544
thread0::fc                          2000        35.9336     0.01663     0.157915    0.0179668   0.15241
thread0::sequence_enumerate          12000       32.7695     0.00238     0.399521    0.0027308   0.13899
thread0::sum                         2000        16.819      0.006673    0.132981    0.00840951  0.0713368
thread0::softsign                    2000        7.47443     0.003436    0.007791    0.00373722  0.0317023
thread0::cos_sim                     1000        7.25779     0.006938    0.084495    0.00725779  0.0307834
```
- after:
```
Event                                Calls       Total       Min.        Max.        Ave.        Ratio.
thread0::hash                        12000       70.4627     0.004945    0.030166    0.0058719   0.311263
thread0::fused_embedding_seq_pool    16000       61.9528     0.003069    0.244838    0.00387205  0.273672
thread0::fc                          2000        31.7109     0.014505    0.434295    0.0158554   0.14008
thread0::sequence_enumerate          12000       31.3849     0.002342    0.023565    0.00261541  0.13864
thread0::sum                         2000        15.8969     0.006678    0.083963    0.00794845  0.0702232
thread0::cos_sim                     1000        7.58994     0.006897    0.356685    0.00758994  0.033528
thread0::softsign                    2000        7.37836     0.00335     0.031451    0.00368918  0.0325933
```

### CODE REVIEW

File: paddle/fluid/operators/fc_op.cc [hunk 1]
```diff
@@ -142,13 +139,20 @@ class FCOpKernel : public framework::OpKernel<T> {
...
     PADDLE_ENFORCE(platform::is_cpu_place(ctx.GetPlace()),
                    "It must use CPUPlace.");
-    auto input = ctx.Input<Tensor>("Input");
-    auto w = ctx.Input<Tensor>("W");
-    auto bias = ctx.Input<Tensor>("Bias");
-    auto output = ctx.Output<Tensor>("Out");
+    auto input = ctx.Input<framework::LoDTensor>("Input");
+    auto w = ctx.Input<framework::LoDTensor>("W");
+    auto bias = ctx.Input<framework::LoDTensor>("Bias");
+    auto output = ctx.Output<framework::LoDTensor>("Out");
```
Comment (tensor-tang) [inline paddle/fluid/operators/fc_op.cc:142]: why change to LodTensor

File: paddle/fluid/operators/fc_op.cc [hunk 2]
```diff
@@ -128,6 +119,12 @@ void FCOpMaker::Make() {
   AddAttr<bool>("use_mkldnn",
                 "(bool, default false) Only used in mkldnn kernel")
       .SetDefault(false);
+  AddAttr<bool>(
+      framework::kAllKernelsMustComputeRuntimeShape,
+      "If an Op has this attribute, all its kernels should calculate output"
```
Comment (tensor-tang) [inline paddle/fluid/operators/fc_op.cc]: Maybe it's not necessary adding these comments,  all op adding these should be duplicated, or comments should be simple.

File: paddle/fluid/operators/fc_op.cc (hunk 1 above)
Comment (luotao1) [reply paddle/fluid/operators/fc_op.cc:142]: If not change to LodTensor, `output->Resize` and `output->set_lod` could not be used.

File: paddle/fluid/operators/fc_op.cc (hunk 2 above)
Comment (luotao1) [reply paddle/fluid/operators/fc_op.cc]: Done, simplify the comments.

File: paddle/fluid/operators/fc_op.cc (hunk 1 above)
Comment (luotao1) [reply paddle/fluid/operators/fc_op.cc:142]: `w` and `bias` already roll back to use `Tensor`.

**Review by tensor-tang — APPROVED** [review]
LGTM

**Pull request merged**